Content.Server/Traits/Assorted/ForeignerTraitSystem.cs

```diff
@@ -0,0 +1,105 @@
+using System.Linq;
+using Content.Server.Hands.Systems;
+using Content.Server.Language;
+using Content.Server.Storage.EntitySystems;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Language;
+using Content.Shared.Language.Components;
+using Content.Shared.Language.Components.Translators;
+using Content.Shared.Storage;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Traits.Assorted;
+
+
+public sealed partial class ForeignerTraitSystem : EntitySystem
+{
+    [Dependency] private readonly EntityManager _entMan = default!;
+    [Dependency] private readonly HandsSystem _hands = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly LanguageSystem _languages = default!;
+    [Dependency] private readonly StorageSystem _storage = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ForeignerTraitComponent, ComponentInit>(OnSpawn); // TraitSystem adds it after PlayerSpawnCompleteEvent so it's fine
+    }
+
+    private void OnSpawn(Entity<ForeignerTraitComponent> entity, ref ComponentInit args)
+    {
+        if (entity.Comp.CantUnderstand && !entity.Comp.CantSpeak)
+            Log.Warning($"Allowing entity {entity.Owner} to speak a language but not understand it leads to undefined behavior.");
+
+        if (!TryComp<LanguageKnowledgeComponent>(entity, out var knowledge))
+        {
+            Log.Warning($"Entity {entity.Owner} does not have a LanguageKnowledge but has a ForeignerTrait!");
+            return;
+        }
+
+        var alternateLanguage = knowledge.SpokenLanguages.Find(it => it != entity.Comp.BaseLanguage);
+        if (alternateLanguage == null)
+        {
+            Log.Warning($"Entity {entity.Owner} does not have an alternative language to choose from (must have at least one non-GC for ForeignerTrait)!");
+            return;
+        }
+
+        if (TryGiveTranslator(entity.Owner, entity.Comp.BaseTranslator, entity.Comp.BaseLanguage, alternateLanguage, out var translator))
+        {
+            _languages.RemoveLanguage(entity, entity.Comp.BaseLanguage, entity.Comp.CantSpeak, entity.Comp.CantUnderstand, knowledge);
+        }
+    }
+
+    /// <summary>
+    ///     Tries to create and give the entity a translator to translator that translates speech between the two specified languages.
+    /// </summary>
```

Comment:
It may be worth discussing, but I just thought that its an option to make it so that this item can be bought in your loadouts if you have the trait. But that may also fuck over players who aren't aware that they don't spawn with the item.